test/cli/harness/harness.go

```diff
@@ -20,42 +19,35 @@ type Harness struct {
 	Runner    *Runner
 	NodesRoot string
 	Nodes     Nodes
+	Log       *TestLogger
+	T         *testing.T
 }
 
-// TODO: use zaptest.NewLogger(t) instead
-func EnableDebugLogging() {
-	err := logging.SetLogLevel("testharness", "DEBUG")
-	if err != nil {
-		panic(err)
+// New constructs a harness that cleans up after the given test is done.
+func New(t *testing.T, options ...func(h *Harness)) *Harness {
+	log := NewTestLogger(t)
+	h := &Harness{
+		Runner: &Runner{Log: log, Env: osEnviron()},
+		Log:    log,
+		T:      t,
 	}
-}
-
-// NewT constructs a harness that cleans up after the given test is done.
-func NewT(t *testing.T, options ...func(h *Harness)) *Harness {
-	h := New(options...)
-	t.Cleanup(h.Cleanup)
-	return h
-}
-
-func New(options ...func(h *Harness)) *Harness {
-	h := &Harness{Runner: &Runner{Env: osEnviron()}}
 
 	// walk up to find the root dir, from which we can locate the binary
 	wd, err := os.Getwd()
 	if err != nil {
-		panic(err)
+		h.Log.Fatal(err)
```

Comment:
should we use t.Fatal when initializing Harness?

Reply:
Using the logger in this case is functionally equivalent. The reason we use it here is for consistency (code, output format, etc.).